components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -486,42 +487,42 @@ export class TypeORMWorkspaceDBImpl extends TransactionalDBImpl<WorkspaceDB> imp
         cutOffDate: Date = new Date(),
         limit: number = 100,
         type: WorkspaceType = "regular",
-    ): Promise<WorkspaceAndOwner[]> {
+    ): Promise<WorkspaceOwnerAndDeletionEligibility[]> {
         // we do not allow to run this with a future date
         if (cutOffDate > new Date()) {
             throw new Error("cutOffDate must not be in the future, was: " + cutOffDate.toISOString());
         }
         const workspaceRepo = await this.getWorkspaceRepo();
-        const dbResults = await workspaceRepo.query(
```

Comment:
I wanted to align this with `findWorkspacesForPurging` by converting from a raw SQL statement, but maybe it's too out of scope here. I'm curious if anyone has thoughts about it

Reply:
Looking at the "async predicate" which looks way too much like a join in disguise, let' snot do this right now. :+1: :upside_down_face:

Reply:
WDYT about the readability of this with [`13e1a76` (#20271)](https://github.com/gitpod-io/gitpod/pull/20271/commits/13e1a76ae9f821d8d29bf84d43e501f11a906daf)? I tried putting everything into a single query and think that it makes a bit more sense.

The benefit of using this join is not too big though so I'm happy to revert if you think it's not that helpful.